data_preparation/global_AgERA5/download_agera5.py

```diff
@@ -0,0 +1,123 @@
+import os
+import cdsapi
+import concurrent.futures
+
+# Most of the code here comes from Abdelrahman's library.
+# https://github.com/AbdelrahmanAmr3/earthstat
+
+AgERA5_params = {
+    "Maximum_Temperature": ("2m_temperature", "24_hour_maximum"),
+    "Minimum_Temperature": ("2m_temperature", "24_hour_minimum"),
+    "Mean_Temperature": ("2m_temperature", "24_hour_mean"),
+    "Solar_Radiation_Flux": ("solar_radiation_flux", None),
+    "Precipitation_Flux": ("precipitation_flux", None),
+}
+
+
+# @author: Abdelrahman Saleh
+def get_agera5_params(sel_param, start_year, end_year, area=None):
+    if sel_param in AgERA5_params:
+        var, stat = AgERA5_params[sel_param]
+
+        # Construct the return dictionary
+        retrieve_params = {
+            "version": "1_1",
+            "format": "zip",
+            "variable": var,
+            "year": [str(year) for year in range(start_year, end_year + 1)],
```

Comment:
"year": [str(year) for year in range(start_year, end_year + 1)] -> "year": str(year)